header.php

```diff
@@ -1,588 +1,593 @@
+} //isset($_POST['sort'])
+/////////////////////////////Filter products based on category/////////////////////////////
+if (isset($_POST['categoryvalues'])) {
+		$CATEGORY_VALUES   = $_POST['categoryvalues'];
+		$CATEGORY_SELECTED = implode(",", $CATEGORY_VALUES);
+		$BIKE_TYPE                = $_SESSION['bikeselected'];
+		if ($BIKE_TYPE == "") {
+				$select_specific_products = "SELECT tblproducts.ProductID, tblproducts.ProductName, tblproducts.ProductPrice, tblproducts.ProductImage, tblproducts.ProductBikeType, tblproducts.ProductDiscountAmount, tblcategory.CategoryName FROM `tblproducts` INNER JOIN tblcategory ON tblproducts.CategoryID=tblcategory.CategoryID WHERE tblcategory.CategoryName = '" . $CATEGORY_SELECTED . "' ORDER BY " . $SORT_COLUMN . " " . $SORT_TYPE . "";
+				$result_products          = mysqli_query($con, $select_specific_products);
+				$CATEGORY                 = $CATEGORY_SELECTED;
+		} //$BIKE_TYPE == ''
+		else {
+				$select_specific_products2 = "SELECT tblproducts.ProductID, tblproducts.ProductName, tblproducts.ProductPrice, tblproducts.ProductImage, tblproducts.ProductBikeType, tblproducts.ProductDiscountAmount, tblcategory.CategoryName FROM `tblproducts` INNER JOIN tblcategory ON tblproducts.CategoryID=tblcategory.CategoryID WHERE tblproducts.ProductBikeType = '" . $BIKE_TYPE . "' AND tblcategory.CategoryName = '" . $CATEGORY_SELECTED . "' ORDER BY " . $SORT_COLUMN . " " . $SORT_TYPE . "";
+				$result_products           = mysqli_query($con, $select_specific_products2);
+				$CATEGORY                  = $CATEGORY_SELECTED;
+		}
+} //isset($_POST['categoryvalues'])
+/////////////////////////////Select All products/////////////////////////////
+else {
+		if ($BIKE_TYPE == '') {
+				$select_all_product1 = "SELECT * FROM `tblproducts`";
+				$result_products     = mysqli_query($con, $select_all_product1);
+				$MESSAGE             = "All Available Products!";
+		} //$BIKE_TYPE == ''
+		else {
+				$select_all_product2 = "SELECT * FROM `tblproducts` WHERE ProductBikeType = '" . $BIKE_TYPE . "'";
+				$result_products     = mysqli_query($con, $select_all_product2);
+				$MESSAGE             = "All Available Products of the Selected Bike!";
+		}
+}
+/////////////////////////////Search Products////////////////////////////
+if (isset($_POST['search'])) {
+	$BIKE_TYPE = $_SESSION['bikeselected'];
+	$SEARCHED_VALUE = $_POST['searchproduct'];
+
+	$search_query = "SELECT * FROM `tblproducts` WHERE CONCAT(`ProductName`) LIKE '%" . $SEARCHED_VALUE . "%'";
+	if ($BIKE_TYPE !== '') {
+		$search_query .= " AND ProductBikeType = '" . $BIKE_TYPE . "'";
+	}
+
+	$result_products = mysqli_query($con, $search_query);
+	$ROWS = mysqli_num_rows($result_products);
+
+	if ($ROWS > 0) {
+		$MESSAGE = "We have found ('" . $ROWS . "') that matches:'" . $SEARCHED_VALUE . "' ";
+		$CATEGORY_SELECTED = '5';
+	} else {
+		$MESSAGE = "No matching products found.";
+		$CATEGORY_SELECTED = ''; // Reset the category selection if no results found
+	}
+}

```

Comment:
Combined the two similar queries into a single query using conditional logic.Removed unnecessary string concatenation and assignment for the `$MESSAGE` variable.Added an additional check for the number of rows to provide appropriate feedback.Added a condition to reset the  `$CATEGORY_SELECTED `variable when no results are found.